src/main/java/org/zowe/sample/apiservice/wto/WtoController.java

```diff
@@ -44,8 +48,15 @@ public WtoController(Wto wto) {
     @ApiOperation(value = "Executes WTO on z/OS and returns a greeting for the name passed", nickname = "greetingToSomeone", authorizations = {
             @Authorization(value = DOC_SCHEME_BASIC_AUTH) })
     @GetMapping("/wto")
-    public WtoDto greeting(
-            @ApiParam(value = "Person or object to be greeted", required = false) @RequestParam(value = "name", defaultValue = "world") String name) {
-        return wto.call(counter.incrementAndGet(), String.format(template, name));
+    public WtoResponse greeting(
+            @ApiParam(value = "Person or object to be greeted", required = false) @RequestParam(value = "name", defaultValue = "world") String name)
+            throws Exception {
+        return (WtoResponse) platformThreadLevelSecurity
```

Comment:
@plavjanik - should I remove the context switch here?

Reply:
yes, please. The server user ID should issue WTO messages